tests/testthat/test-data_modify.R

```diff
@@ -492,6 +492,18 @@ test_that("data_modify works with functions that return character vectors", {
 })
 
 
+test_that("data_modify 1:n() and similar works in grouped data frames", {
```

Comment:
Can you also check how it behaves on 

1. ungrouped dataframe
2. dataframe with 0 rows
3. maybe also inside a function to check that the `eval()` call runs in the right environment?
4. calls that are close but not identical to `n()`, e.g. `fun()`, `1:fun()`, etc.

Reply:
All checks added, also behaviour for zero-row data frames.
But I have no good idea for a use-case for `fun()`?